File: packages/sdk-platforms/rust/zksync-sso/crates/sdk/src/utils/session/session_lib/session_state/limit_state.rs
use crate::{
    contracts::SessionLib::LimitState as SessionLibLimitState,
    utils::alloy::serde_helpers::{
        deserialize_u256_from_integer_string, serialize_u256_as_integer_string,
    },
};
use alloy::primitives::{Address, FixedBytes, U256};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub struct LimitState {
    #[serde(
        serialize_with = "serialize_u256_as_integer_string",
        deserialize_with = "deserialize_u256_from_integer_string"
    )]
    pub remaining: U256,
    pub target: Address,
    pub selector: FixedBytes<4>,
    #[serde(
        serialize_with = "serialize_u256_as_integer_string",
        deserialize_with = "deserialize_u256_from_integer_string"
    )]
    pub index: U256,
}

impl From<SessionLibLimitState> for LimitState {
    fn from(value: SessionLibLimitState) -> Self {
        LimitState {
            remaining: value.remaining,
            target: value.target,
            selector: value.selector,
            index: value.index,
        }
    }
}

File: packages/sdk-platforms/swift/ZKsyncSSO/Sources/ZKsyncSSO/Account/Sessions/SessionLib/SessionState/LimitState.swift
import Foundation

public struct LimitState: Codable, Equatable, Hashable, Sendable {
    public var remaining: String
    public var target: String
    public var selector: String
    public var index: String
    
    public init(
        remaining: String,
        target: String,
        selector: String,
        index: String
    ) {
        self.remaining = remaining
        self.target = target
        self.selector = selector
        self.index = index
    }
}